### DISCUSSION TITLE
SARIF

### DISCUSSION THREAD

**Discussion by Bananeweizen** [ota4j-team/open-test-reporting#2] [Ideas]
Did you have a look at [SARIF](https://github.com/microsoft/sarif-tutorials) during creation of this report format? From a (very) simplified point of view, static analysis issues and test reports have some commonalities for developers that have to deal with them. From a more sophisticated point of view however, I must admit it seems hard to find a common technical ground.
But maybe there are things to be shared between the communities beyond technical stuff, e.g. how to make tool vendors actually adopt the new standard.

**Comment by marcphilipp** [comment]
I took a look but I didn't see that SARIF is able to represent hierarchical structures. Am I missing something?

**Reply by Bananeweizen** [reply]
Honestly, I myself don't know much more than the existence of the format. I had stumbled over it multiple times in the last years when looking into Jenkins reporting plugins, tools like checkstyle and Sonarqube etc. I mainly noticed how long it took to be adopted in at least some tools, and I just wish the open reporting format to become more popular more quickly.

Thanks for investigating, but I guess there is no use in digging any deeper then.
